File: src/java/bps-data-generator/src/test/java/com/github/rnowling/bps/datagenerator/generators/transaction/TestCustomerTransactionParametersGenerator.java
package com.github.rnowling.bps.datagenerator.generators.transaction;

import static org.junit.Assert.assertTrue;

import org.junit.Test;

import com.github.rnowling.bps.datagenerator.Constants;
import com.github.rnowling.bps.datagenerator.SeedFactory;

public class TestCustomerTransactionParametersGenerator
{

	@Test
	public void testGenerator() throws Exception
	{
		SeedFactory seedFactory = new SeedFactory(1234);
		CustomerTransactionParametersGenerator generator = new CustomerTransactionParametersGenerator(seedFactory);
		
		CustomerTransactionParameters transParams = generator.generate();
		
		assertTrue(transParams.countPets() >= Constants.MIN_PETS);
		assertTrue(transParams.countPets() <= Constants.MAX_PETS);
		assertTrue(transParams.getAveragePurchaseTriggerTime() >= Constants.PURCHASE_TRIGGER_TIME_MIN);
		assertTrue(transParams.getAveragePurchaseTriggerTime() <= Constants.PURCHASE_TRIGGER_TIME_MAX);
		assertTrue(transParams.getAverageTransactionTriggerTime() >= Constants.TRANSACTION_TRIGGER_TIME_MIN);
		assertTrue(transParams.getAverageTransactionTriggerTime() <= Constants.TRANSACTION_TRIGGER_TIME_MAX);
	}

}

File: src/python/tests/generators/transaction_generators/test_customer_transaction_parameters_generator.py
import unittest

import simulation_parameters as sim_param

from generators.transaction_generator.customer_transaction_parameters_generator import CustomerTransactionParameters
from generators.transaction_generator.customer_transaction_parameters_generator import CustomerTransactionParametersGenerator

class TestCustomerTransactionParametersGenerator(unittest.TestCase):
    def test_generate(self):
        generator = CustomerTransactionParametersGenerator()
        
        trans_params = generator.generate()

        self.assertIsInstance(trans_params, CustomerTransactionParameters)
        self.assertIsInstance(trans_params.pet_counts, dict)

        total_count = sum(trans_params.pet_counts.values())
        self.assertTrue(total_count >= sim_param.MIN_PETS)
        self.assertTrue(total_count <= sim_param.MAX_PETS)

        self.assertTrue(trans_params.average_transaction_trigger_time >= sim_param.TRANSACTION_TRIGGER_TIME_MIN)
        self.assertTrue(trans_params.average_transaction_trigger_time <= sim_param.TRANSACTION_TRIGGER_TIME_MAX)

        self.assertTrue(trans_params.average_purchase_trigger_time >= sim_param.PURCHASE_TRIGGER_TIME_MIN)
        self.assertTrue(trans_params.average_purchase_trigger_time <= sim_param.PURCHASE_TRIGGER_TIME_MAX)
